### PR TITLE
Welding tools can be used to light other people's cigarettes

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Allows welding tools to light another person's cigarette by targeting the mouth whilst there is an unlit cigarette perched between their lips.

## Why It's Good For The Game
When someone asks you if you have a light, you can nod with confidence as you pull out your trusty welder like a true badass.

## Testing
A number of skrell were given cigarettes. They then lit each other's cigarettes with a welding tool that was also provided.

No skrell were harmed in the making of this PR.

## Changelog
:cl:
add: Unlit cigarettes in the mouth slot can be lit by another person using a lit welder when targeting the mouth.
/:cl:

### CODE REVIEW

File: code/game/objects/items/tools/welder.dm [hunk 1]
```diff
@@ -146,6 +146,18 @@
 		return
 	remove_fuel(0.5)
 
+/obj/item/weldingtool/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
+	if(istype(M.wear_mask, /obj/item/clothing/mask/cigarette) && user.zone_selected == "mouth" && tool_enabled) // For lighting other people's cigarettes.
```
Comment (lewcc) [inline code/game/objects/items/tools/welder.dm:150]: Needs a sanity check
```suggestion
/obj/item/weldingtool/attack(mob/living/carbon/M, mob/living/carbon/user)
	if(istype(M) && istype(M.wear_mask, /obj/item/clothing/mask/cigarette) && user.zone_selected == "mouth" && tool_enabled) // For lighting other people's cigarettes.
```

**Comment by Probablyjay19** [comment]
Almost 100% positive you can already do this by opening up the strip menu and using the welder on the strip option for the cigarette, but this is simpler. Bravo.

**Comment by Fordoxia** [comment]
> Almost 100% positive you can already do this by opening up the strip menu and using the welder on the strip option for the cigarette, but this is simpler. Bravo.

You cannot. Trying this will simply attempt to strip the cigarette.

File: code/game/objects/items/tools/welder.dm [hunk 2]
```diff
@@ -146,6 +146,18 @@
...
+	if(istype(M) && istype(M.wear_mask, /obj/item/clothing/mask/cigarette) && user.zone_selected == "mouth" && tool_enabled) // For lighting other people's cigarettes.
+		var/obj/item/clothing/mask/cigarette/cig = M.wear_mask
+		if(M == user)
+			cig.attackby(src, user)
+		else
+			cig.light("<span class='notice'>[user] holds out [src] out for [M], and casually lights [cig]. What a badass.</span>")
+			playsound(src, 'sound/items/lighter/light.ogg', 25, TRUE)
+			M.update_inv_wear_mask()
+	else
+		..()
```
Comment (S34NW) [inline code/game/objects/items/tools/welder.dm:159]:
```suggestion
	// For lighting other people's cigarettes.
	var/obj/item/clothing/mask/cigarette/cig = M?.wear_mask
	if(!istype(cig) || user.zone_selected =! "mouth" || !tool_enabled) 
		..()

	if(M == user)
		cig.attackby(src, user)
		return

	cig.light("<span class='notice'>[user] holds out [src] out for [M], and casually lights [cig]. What a badass.</span>")
	playsound(src, 'sound/items/lighter/light.ogg', 25, TRUE)
	M.update_inv_wear_mask()

```

File: code/game/objects/items/tools/welder.dm [hunk 3]
```diff
@@ -146,6 +146,21 @@
 		return
 	remove_fuel(0.5)
 
+/obj/item/weldingtool/attack(mob/living/carbon/M, mob/living/carbon/user)
+	// For lighting other people's cigarettes.
+	var/obj/item/clothing/mask/cigarette/cig = M?.wear_mask
+	if(!istype(cig) || user.zone_selected =! "mouth" || !tool_enabled)
```
Comment (Fordoxia) [inline code/game/objects/items/tools/welder.dm:152]:
```suggestion
	if(!istype(cig) || user.zone_selected != "mouth" || !tool_enabled) 
```

**Review by S34NW — APPROVED** [review]

File: code/game/objects/items/tools/welder.dm [hunk 4]
```diff
@@ -146,6 +146,21 @@
...
+
+	if(M == user)
+		cig.attackby(src, user)
+		return
+
+	cig.light("<span class='notice'>[user] holds out [src] out for [M], and casually lights [cig]. What a badass.</span>")
+	playsound(src, 'sound/items/lighter/light.ogg', 25, TRUE)
+	M.update_inv_wear_mask()
+
+
```
Comment (DGamerL) [inline code/game/objects/items/tools/welder.dm:163]:
```suggestion
```

**Pull request merged**